### PR TITLE
Transformer step

### PR DESCRIPTION
**Pull request by Robert-Kolmos**
Requires #18.
- Implement the section transformer step, and add unit tests.
- Refactor the ResourceTaskRespository.

### CODE REVIEW

File: domain/src/main/java/org/sagebionetworks/research/domain/repository/TaskRepository.java [hunk 1]
```diff
@@ -36,27 +36,66 @@
...
 
+    /**
+     * Gets the task result with the given UUID.
+     * @param taskRunUUID The UUID of the result to get.
+     * @return the task result with the given UUID.
+     */
     @NonNull
     Maybe<TaskResult> getTaskResult(UUID taskRunUUID);
 
+    /**
```
Comment (liujoshua) [inline domain/src/main/java/org/sagebionetworks/research/domain/repository/TaskRepository.java]: IMO these don't need to be on the interface. Other implementations of TaskRepository could implement it differently

File: domain/src/main/java/org/sagebionetworks/research/domain/form/ChoiceBase.java [hunk 2]
```diff
@@ -165,43 +165,42 @@ protected boolean equalsHelper(Object o) {
...
                     throws JsonParseException {
                 if (json.isJsonObject()) {
-                    // Suppose we are deserializing a Choice<T> for some type T. We create a TypeToken<ChoiceBase<T>>
-                    // then we deserialize using this.
-                    TypeToken token = createChoiceBaseTypeToken(typeOfT);
-                    ChoiceBase<?> result = context.deserialize(json, token.getType());
-                    if (result != null) {
-                        return result;
-                    }
+                    JsonObject object = json.getAsJsonObject();
```
Comment (liujoshua) [inline domain/src/main/java/org/sagebionetworks/research/domain/form/ChoiceBase.java]: Does Gson's default behavior not work here?

Comment (Robert-Kolmos) [reply domain/src/main/java/org/sagebionetworks/research/domain/form/ChoiceBase.java]: The default behavior would work except for the thing where { "value":"string", "text":"string" } and "string" should deserialize into the same ChoiceBase. If we removed support for that I believe I could just use Gson's default behavior. What are your thoughts on doing that?

File: data/src/main/java/org/sagebionetworks/research/data/ResourceTaskRepository.java [hunk 3]
```diff
@@ -131,24 +131,38 @@ private InputStreamReader getAsset(String assetPath) throws IOException {
...
 
+    /**
+     * Returns and InputStreamReader corresponding to the given task asset.
+     * @param assetName The name of the asset to get.
+     * @return An InputStreamReader corresponding to the Task asset with the given name.
+     * @throws IOException If the asset of the given name cannot be opened.
+     */
     @NonNull
-    @Override
     public InputStreamReader getJsonTaskAsset(String assetName) throws IOException {
```
Comment (Robert-Kolmos) [inline data/src/main/java/org/sagebionetworks/research/data/ResourceTaskRepository.java]: Should this be public or private?

**Pull request merged**